src/formatDate.js

```diff
@@ -50,7 +50,65 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let oldSeparator = '';
+  let newSeparator = '';
+  let oldYearIndex = '';
+  let oldYearLength = '';
+  let oldMonthIndex = '';
+  let oldDayIndex = '';
+  const newDate = [];
+
+  for (const itemFrom of fromFormat) {
+    if (itemFrom.includes('Y')) {
```

Comment:
You can use [switch...case](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) to provide all the checks properly